frontend/src/search.js

```diff
@@ -83,6 +83,34 @@ search.addWidgets([
         document.getElementById('under-progress').classList.remove('hidden');
     },
   },
+  {
+    message: '',
+    init() {
+      document.getElementById('retry-search').addEventListener('click', () => {
+        document.getElementById('loading').classList.remove('hidden');
+        document.getElementById('stats').classList.add('hidden');
+        document.getElementById('error').classList.add('hidden');
+        this.message = '';
+        setTimeout(() => {
+          search.helper.search();
+        }, 500);
+      });
+      search.on('error', ({ error }) => {
```

Comment:
I doubt we need to hook this event within a widget. Please see if we can get all the necessary information from the `searchMetadata` in the `render` method.

Reply:
`searchMetadata` does not include any information about an error.

Reply:
This discussion is related to the [other thread](https://github.com/PureBhaktiArchive/archive-web-app/pull/120/files/b69b15307729d874105b650bc48fff86bdb0361b#r683480695). Let's proceed there.